### DISCUSSION TITLE
Installation failure vSphere 6.5 and 6.7

### DISCUSSION THREAD

**Discussion by lerou114** [Security-Onion-Solutions/securityonion#2217] [General] [answered]
I am currently attempting to deploy a POC/Test instance 2.3.10 in VMWare after multiple attempts on our VMware infrastructure I am having multiple failures. I am able to install successfully on a workstation with Virtualbox.

I have attempted the failed on installs on VMWare 6.5 &  6.7
VM:
6 Cores
16 GB RAM
240 GB Storage

Version 2.3.10
Security onion ISO and Network installation Ubuntu and CentOS
Intall types tried: Standalone and Manager/Search Node

Install had a problem. Please see /root/sosetup.log for details.
so-status so-mysql missing.

local: 
 Data failed to compile:
-----------
	The function "state.highstate" is runnning as PID 17976 and was started at 2020, Dec 04 15:38:20.829566 with jid 
20201204153820829566

I have pulled these snippets from the so.setup.log file I have removed the ip preceeding dial tcp.

[INFO    ] Executing command 'yum check-update 2>&1 | grep "Error: rpmdb open failed"' in directory '/root'

[+] Set the address config key to "https://localhost:8080" in the "default" context
[+] Set the tls-skip-verify config key to "true" in the "default" context
[+] Set the url-prefix config key to "/fleet" in the "default" context
No username supplied, using email as username
error setting up Fleet: POST /api/v1/setup: Post "https://localhost:8080/fleet/api/v1/setup": dial tcp :8080: connect: connection timed out
Please log in with: fleetctl login
Please log in with: fleetctl login
Enabling Fleet...
[ERROR   ] Command '['docker', 'exec', 'so-fleet', 'fleetctl', 'get', 'enroll-secret', 'default']' failed with return code: 1
[ERROR   ] stdout: Please log in with: fleetctl login
[ERROR   ] retcode: 1
[ERROR   ] Command 'docker exec so-fleet fleetctl get enroll-secret default' failed with return code: 1
[ERROR   ] output: Please log in with: fleetctl login
[CRITICAL] Rendering SLS 'base:fleet.event_enable-fleet' failed: mapping values are not allowed in this context
Generating osquery install packages - this will take some time...
Installing launcher via salt...
so-nginx

Total run time:   234.088 s
Errors detected during setup; skipping post-setup steps to allow for analysis of failures.

**Comment by defensivedepth** [comment]
Can you post the full sosetup.log somewhere?

**Reply by lerou114** [reply]
[sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/5644272/sosetup.log)

**Reply by defensivedepth** [reply]
Can you post /root/fleet-setup.log?

Are you changing any of the default settings when you run through setup?

**Reply by lerou114** [reply]
I am taking all the defaults in the wizard. The only thing I have changed is that I added our Class B address space as the home network. This is the log with the ip address of the host stripped out. 

[fleet-setup.log](https://github.com/Security-Onion-Solutions/securityonion/files/5645682/fleet-setup.log)

**Reply by defensivedepth** [reply]
Last question, I think :)

If you still have the failed install online, can you post a sanitized version of this: (they keys are fine to leave, those are generated every time sosetup is run)

`sudo salt-call pillar.get global`

Is `fleet_ip` the same as `managerip` ?

Also:

`sudo salt-call grains.get ip_interfaces` - you can do x.x.x.x

**Reply by lerou114** [reply]
No worries! 
[saltgrain.txt](https://github.com/Security-Onion-Solutions/securityonion/files/5645831/saltgrain.txt)
[saltpillars.txt](https://github.com/Security-Onion-Solutions/securityonion/files/5645833/saltpillars.txt)

**Reply by lerou114** [reply]
The fleet_ip and the managerip are the same.

**Reply by defensivedepth** [reply]
I just ran a Manager + Search 2.3.10 ISO install and compared the logs & config to what you posted - I am not seeing the same issue.

Do you have any logs from the Network install? At this point, I am trying to see if the install is failing at the same point whether it's ISO or Network Install, etc.

**Reply by lerou114** [reply]
So my colleague and I ran both the net install and cd install on hardware and duplicated the same scenario this also happened with a bare metal install. However after running so-allow for an analyst the install started working. The web ui was available in the beginning but fleet page was not working. However after period of time the fleet login started working. If that makes sense.

**Reply by lerou114** [reply]
Net Install logs:
[sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/5674900/sosetup.log)
[pillar.log](https://github.com/Security-Onion-Solutions/securityonion/files/5674902/pillar.log)
[grains.log](https://github.com/Security-Onion-Solutions/securityonion/files/5674903/grains.log)

**Reply by defensivedepth** [reply]
Thanks, will look this over in the next day

**Reply by lerou114** [reply]
So I am not well as well versed as I should be in using docker. On my install I do not have the default fleet secret. Is there a way to generate the enroll_secret that is missing?

**Reply by defensivedepth** [reply]
What output do you get when you run the following on the Manager?

`salt-call pillar.get secrets:fleet_enroll-secret`

**Reply by lerou114** [reply]
local:
     None

**Reply by defensivedepth** [reply]
In all of the sosetups that you have posted ( including the most recent one), it appears that Fleet failed to install because it is resolving `https://localhost:8080/fleet/api/v1/setup":` to something other than 127.0.0.1 (Couldn't confirm that until you posted the most recent sosetup)..... which is really odd.

Can you run the following on the manager and post the output?

`sudo docker exec  so-fleet cat /etc/hosts`

**Reply by lerou114** [reply]
127.0.0.1       localhost
::1     localhost ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
172.17.0.19     so-fleet

**Comment by defensivedepth** [comment, marked as answer]
We were able to pin down the problem to the fact that when the Fleet container was resolving `localhost` to something other than 127.0.0.1.

This was fixed with swapping localhost for 127.0.0.1 - this fix will be release in 2.3.20.

https://github.com/Security-Onion-Solutions/securityonion/pull/2346
